src/nlp/api.jl

```diff
@@ -1298,6 +1299,41 @@ function hess_op!(
   return LinearOperator{T}(nlp.meta.nvar, nlp.meta.nvar, true, true, prod!, prod!, prod!)
 end
 
+"""
+    P = tensor_projection(nlp, n, x, directions, args...)
+
+Returns the projection of the n-th derivative of the objective of `nlp` at `x` along the specified directions.
```

Comment:
What about `tensor_projection` ?
You should also have
`P = tensor_projection(nlp, n, x, y, directions, args...)`
as the default, where by default
`P = tensor_projection(nlp, n, x, directions, args...)`
call the first one with zeros(n).

Reply:
I agree that it will be great to have tensor products with both the objective and the Lagrangian.
Should we also add an API for just the constraints, or include a scaling factor `\alpha` that can go in front of the objective?

Reply:
I think it makes sense to be coherent with the other functions like `hprod`.